lib/src/collection/treeset.dart

```diff
@@ -627,10 +625,10 @@ class AvlTreeSet<V> extends TreeSet<V> {
    * See [Set.retainAll]
    */
   void retainAll(Iterable<Object> elements) {
-    List<V> chosen = [];
+    List<V> chosen = <V>[];
     for (var target in elements) {
       if (contains(target)) {
-        chosen.add(target);
+        chosen.add(target as V);
```

Comment:
There's active work on making the core libraries strong mode friendly, please file any issues or requests here:  https://github.com/dart-lang/sdk/issues .  area-library is the right tag.